tests/users/test_api_login.py

```diff
@@ -43,6 +46,54 @@ def setUp(self):
         db.session.add(self.first_user)
         db.session.add(self.second_user)
         db.session.commit()
+    def test_user_login_invalid_credentials(self):
+        with self.client:
+
+            response = self.client.post(
+                "/login",
+                data=json.dumps(
+                    dict(username="invaliduser", password="invalidpassowrd")
+                ),
+                follow_redirects=True,
+                content_type="application/json",
+            )
+
+            self.assertIsNone(response.json.get("access_token"))
+            self.assertIsNone(response.json.get("access_expiry"))
+            self.assertIsNone(response.json.get("refresh_token"))
+            self.assertIsNone(response.json.get("refresh_expiry"))
+
+            self.assertEqual(1, len(response.json))
+            self.assertEqual(
+                messages.WRONG_USERNAME_OR_PASSWORD, response.json
+            )
+
+            self.assertEqual(HTTPStatus.UNAUTHORIZED, response.status_code)
+    def test_user_login_invalid_credentials(self):
```

Comment:
please add a blank line below. It for readability concerns. Unfortunately, we don't have a lint checker as part of the checks, yet, so I am just being a human linter 🙈

Reply:
oh wait, isn't this test the exact same as the above one? If so, can you please remove it?

Reply:
Oh yes, you are correct. Thanks for checking. I was planning to add separate test cases and I missed it. I am removing the extra code.

Reply:
I have removed the duplicate test case